### DISCUSSION TITLE
Has the development team considered warp a terminal control for use by WinUI?

### DISCUSSION THREAD

**Discussion by Endt4sk** [microsoft/terminal#14987] [Q&A] [answered]
Has the development team considered warp a terminal control for use by WinUI?

**Comment by zadjii-msft** [comment, marked as answer]
I'm under the impression that Warp's renderer and UI stack is macOS only at this time. They're more than welcome to port it to Windows if they'd like. The OSS code here in the Terminal repo should provide more than enough examples of what they might need to do to build a WinUI terminal control (and a WinUI application to host that control), as well as how to use the ConPTY infrastructure on Windows. 

We'll probably continue to maintain our own terminal control, and build our own Warp-like features in parallel. I think there's plenty of room for us to pursue parallel innovation in this space ☺️
